generator/internal/discovery_type_vertex.cc

```diff
@@ -44,6 +44,34 @@ absl::optional<std::string> CheckForScalarType(nlohmann::json const& j) {
   return absl::nullopt;
 }
 
+std::string FormatMessageDescription(nlohmann::json const& field,
+                                     int indent_level) {
+  std::string description;
+  if (field.contains("description")) {
+    description = absl::StrCat(
+        FormatCommentBlock(std::string(field["description"]), indent_level),
+        "\n");
+  }
+
+  if (field.contains("enum")) {
+    std::vector<std::pair<std::string, std::string>> enum_comments;
+    for (unsigned int i = 0; i < field["enum"].size(); ++i) {
+      if (field.contains("enumDescriptions") &&
+          i < field["enumDescriptions"].size()) {
+        enum_comments.emplace_back(std::string(field["enum"][i]),
+                                   std::string(field["enumDescriptions"][i]));
+      } else {
+        enum_comments.emplace_back(std::string(field["enum"][i]),
+                                   std::string());
+      }
```

Comment:
This looks like an example of something I occasionally nit about, where much of the conditional behavior is not conditional.  That makes it harder to read.